src/spurt/io/_slc_stack.py

```diff
@@ -0,0 +1,133 @@
+"""Read test SLC stacks during development."""
+
+from __future__ import annotations
+
+import os
+from pathlib import Path
+
+import numpy as np
+
+from ._raster import Raster
+from ._three_d import Irreg3DInput
+
+__all__ = [
+    "SLCStackReader",
+]
+
+
+class SLCStackReader:
+    """Read SLC stack from single files."""
+
+    def __init__(
+        self,
+        folder_name: str | os.PathLike[str],
+        qthreshold: float = 0.6,
+    ):
+        """Initialize stack by scanning a folder.
+
+        We only care about temporal coherence and int files
+        for now. metdata and coherence to be dealt with later.
+        """
+        # Here are members that we hold by default
+        self.dates: list = []
+        self.files: dict = {
+            "quality": None,
+            "slc": {},
+        }
+        self.raster_shape: tuple = ()
+        self.folder_name = folder_name
+        self.qthreshold = qthreshold
+
+        # Discover and populate the files
+        self._discover()
+
+    def _discover(self) -> None:
+        """Scan folder and set file names."""
+        p = Path(self.folder_name)
+
+        # First get temporal coherence
+        self.files["quality"] = next(p.glob("temporal_coherence.tif")).absolute()
+        # Read in shape
+        with Raster(self.files["quality"]) as fin:
+            self.raster_shape = fin.shape
+
+        # Then list interferograms
+        ifglist = sorted(p.glob("*.int.tif"))
+        first_date = ifglist[0].name.split("_")[0]
+        self.dates.append(first_date)
+
+        for ifg in ifglist:
+            if not ifg.name.startswith(first_date):
+                errmsg = (
+                    f"Error scanning {self.folder_name}. "
+                    f"{ifg.name} does not conform to interferogram"
+                    " naming convention."
+                )
+                raise ValueError(errmsg)
+
+            acq_date = ifg.name.split("_")[1][:8]
+            self.dates.append(acq_date)
+            if acq_date in self.files["slc"]:
+                errmsg = (
+                    f"Error scanning {self.folder_name}. "
+                    f"{acq_date} already in list of SLCs. "
+                    f"{ifg.name} appears to be a duplicate"
+                )
+                raise ValueError(errmsg)
+
+            self.files["slc"][acq_date] = ifg.absolute()
```

Comment:
Are these files SLCs or interferograms? The variable names somehow imply both...

If they're supposed to be interferograms, then maybe we should rename this class to `IgramStackReader` or something similar. If not, then let's choose a new name for `ifg`.

Reply:
Done